docs/articles/baseplatform/booting-qemu-image.rst

```diff
@@ -2,22 +2,81 @@
 
 .. _booting-a-qemu-image:
 
-Booting up a QEMU image
-=======================
+####################################
+Building and booting up a QEMU image
+####################################
+
+Booting a QEMU image can be using one of the two scripts
+
+- ``runqemu``
+- ``qemu_launcher``
+
+Both are scripts but they are offered from different layers. ``runqemu`` is a part of OpenEmbedded while qemu_launcher is a part of meta-pelux. ``qemu_launcher`` requires enabling the virtualization on BIOS while ``runqemu`` is more straight forward and simple. Both options are described below. 
+
+******************************
+Building pelux for QEMU target
+******************************
+
+First we need to have a build. 
+
+On a directory of choice, run:
+
+.. code-block:: bash
+
+  $ repo init - u https://github.com/pelagicore/pelux-manifests
+  $ repo sync
+
+This will create the ``.repo`` and ``sources`` directories. Now we need to create a build directory and source the qemu configuration by running the following command
+
+.. code-block:: bash
+
+  TEMPLATECONF=`pwd`/sources/meta-pelux/conf/variant/qemu-x86-64_nogfx source sources/poky/oe-init-build-env build
+
+Now you can start bitbaking the image, and configure your environment for qemu.
+
+.. code-block:: bash
+
+          # Bitbake core-image-pelux-minimal-dev for qemu-x86-64_nogfx target
+          $ bitbake core-image-pelux-minimal-dev
+
+          # Make sure we have the required utilities
+          $ bitbake qemu-helper-native
+
+          # Set up networking for qemu
+          $ sudo ../sources/poky/scripts/runqemu-gen-tapdevs 1000 1000 4 tmp/sysroots-components/x86_64/qemu-helper-native/usr/bin
+
+          # Give qemu permissions to access tun interfaces
+          $ sudo setcap cap_net_admin+ep $(find tmp/work/ -name qemu-system-x86_64 | grep qemu-helper-native)
+
+The build is done and you are ready to boot up qemu.
+
+Run qemu script
+---------------
+Finally, you can ``runqemu`` 
+
+.. code-block:: bash
+
+          $  runqemu
+
+Now on another terminal of the same machine, you should be able to ssh ``root@192.168.7.2``. 
+
+QEMU launch script
+------------------
+
+``qemu_launcher`` aims for the same goal as ``runqemu`` but is aimed more towards KVM. It wraps up the kvm command and provides simple options to launch dev and prod images without loosing the flexibility to adjust parameters like memory, cpu, etc.
```

Comment:
no need to abbreviate dev and prod

Reply:
I did that change, but how about removing the development and production entirely, since for QEMU we only provide the minimal and minimal-dev images. Also neither one of them is production..

Reply:
Good point! :-)

Reply:
So, I do not really understand why qemu_launcher is needed. As far as I know QEMU uses KVM if it is available (and I think it is available in all popular Linux distributions since... a very long while back).

(Asking this question here since I am not sure adding a section for it makes sense... if it is not needed. :)

Reply:
If the `runqemu` script does not use KVM by default if it is available, one just has to append `kvm` to the command line, it looks like.

Reply:
And `qemu_launcher` does not use QEMU? I don't know... the more I look at it the more confused I get.

Reply:
Commit message nor #97 contains any information as to why it is needed and why it should be used instead of `runqemu`. :(

Reply:
Anyway, not blocking this to get merged. I am just confused and wonder if it really makes sense to have it at all... and if we can't explain to ourselves why it is needed, how can we do so to someone reading this section. :)

Reply:
>As far as I know QEMU uses KVM if it is available.
That happens when u start QEMU with `runqemu`. On the other hand, if you want to use qemu_launcher, you must have KVM. 
>  qemu_launcher does not use QEMU?
It does.. I dont get it at what point did this confuse you?
> Commit message nor #97 contains any information 
Are you sure u meant about #97? It doesnt seem relevant to this PR. 
However, my understanding of why we should use qemu_launcher is that it has been developed internally.

Reply:
Generally speaking tho, i was tempted to remove any information about qemu_launcher, but after some discussion it was decided to only add the information about runqemu and leave the topic of qemu_launcher for advanced users.

Reply:
> > As far as I know QEMU uses KVM if it is available.
> 
> That happens when u start QEMU with `runqemu`. On the other hand, if you want to use qemu_launcher, you must have KVM.
> 

Yes, so what is the point?

> > qemu_launcher does not use QEMU?
> 
> It does.. I dont get it at what point did this confuse you?

Nope, it does `sudo kvm ...` . "QEMU" is only used in help strings etc.

> > Commit message nor #97 contains any information
> 
> Are you sure u meant about #97? It doesnt seem relevant to this PR.

Sorry, the PR is in meta-pelux: https://github.com/Pelagicore/meta-pelux/pull/97 .

Reply:
So you recommending to remove the section about qemu_launcher entirely? I dont really have a strong opinion on that and i would agree that it only confuses the user.